iOS 9: Crash in mbgl::HTTPRequest::start()

Top of master, hitting the assertion in the first line of the function. Here's the stack:
````
#3	0x00000001004c308c in mbgl::HTTPRequest::start() at /Users/admin/Developer/mapbox-gl-native/platform/darwin/http_request_nsurl.mm:152
#4	0x00000001004c7060 in mbgl::HTTPRequest::retry()::$_2::operator()() const at /Users/admin/Developer/mapbox-gl-native/platform/darwin/http_request_nsurl.mm:361
#5	0x00000001004c6e5c in decltype(std::__1::forward<mbgl::HTTPRequest::retry()::$_2&>(fp)(std::__1::forward<>(fp0))) std::__1::__invoke<mbgl::HTTPRequest::retry()::$_2&>(mbgl::HTTPRequest::retry()::$_2&&&) [inlined] at /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__functional_base:413
#6	0x00000001004c6e48 in std::__1::__function::__func<mbgl::HTTPRequest::retry()::$_2, std::__1::allocator<mbgl::HTTPRequest::retry()::$_2>, void ()>::operator()() at /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/functional:1370
#7	0x000000010026fffc in std::__1::function<void ()>::operator()() const at /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/functional:1756
#8	0x00000001004c96dc in uv::timer::timer_cb(uv_timer_s*, int) at /Users/admin/Developer/mapbox-gl-native/src/mbgl/util/uv_detail.hpp:161
#9	0x0000000100548c58 in uv__run_timers at /Users/admin/Developer/mapbox-gl-native/mason_packages/.build/libuv-0.10.28/src/unix/timer.c:146
#10	0x0000000100541e00 in uv_run at /Users/admin/Developer/mapbox-gl-native/mason_packages/.build/libuv-0.10.28/src/unix/core.c:308
#11	0x000000010017f6d4 in uv::loop::run() at /Users/admin/Developer/mapbox-gl-native/src/mbgl/util/uv_detail.hpp:62
#12	0x000000010036e178 in void mbgl::util::Thread<mbgl::DefaultFileSource::Impl>::run<std::__1::tuple<mbgl::FileCache*&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>, 0ul, 1ul>(mbgl::util::ThreadContext, std::__1::tuple<mbgl::FileCache*&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>&&, std::__1::integer_sequence<unsigned long, 0ul, 1ul>) at /Users/admin/Developer/mapbox-gl-native/src/mbgl/util/thread.hpp:125
#13	0x000000010036e02c in mbgl::util::Thread<mbgl::DefaultFileSource::Impl>::Thread<mbgl::FileCache*&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>(mbgl::util::ThreadContext const&, mbgl::FileCache*&&&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&&&)::'lambda'()::operator()() const at /Users/admin/Developer/mapbox-gl-native/src/mbgl/util/thread.hpp:104
#14	0x000000010036dd10 in std::__1::__invoke<mbgl::util::Thread<mbgl::DefaultFileSource::Impl>::Thread<mbgl::FileCache*&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>(mbgl::util::ThreadContext const&, mbgl::FileCache*&&&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&&&)::'lambda'()>(decltype(std::__1::forward<mbgl::util::Thread<mbgl::DefaultFileSource::Impl>::Thread<mbgl::FileCache*&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>(mbgl::util::ThreadContext const&, mbgl::FileCache*&&&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&&&)::'lambda'()>(fp)(std::__1::forward<>(fp0))), mbgl::util::Thread<mbgl::DefaultFileSource::Impl>::Thread<mbgl::FileCache*&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>(mbgl::util::ThreadContext const&, mbgl::FileCache*&&&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&&&)::'lambda'()&&) [inlined] at /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__functional_base:413
#15	0x000000010036dd04 in _ZNSt3__116__thread_executeIZN4mbgl4util6ThreadINS1_17DefaultFileSource4ImplEEC1IJRPNS1_9FileCacheERKNS_12basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEEEEEERKNS2_13ThreadContextEDpOT_EUlvE_JEJEEEvRNS_5tupleIJT_DpT0_EEENS_15__tuple_indicesIJXspT1_EEEE [inlined] at /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/thread:332
#16	0x000000010036dcf4 in std::__1::__thread_proxy<std::__1::tuple<mbgl::util::Thread<mbgl::DefaultFileSource::Impl>::Thread<mbgl::FileCache*&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>(mbgl::util::ThreadContext const&, mbgl::FileCache*&&&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&&&)::'lambda'()> >(void*, void*) at /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/thread:342
````

Running iOS 9 beta 4.

[got access to @quicklywilliam's custom style]

Not sure if it's a factor, but I still don't get the crash. I am seeing a lot of e.g. 

```
[ERROR] {Map}[Style]: Failed to parse [10/292/392]: pbf unknown field type exception
```

--- a/platform/ios/MGLMapView.mm
+++ b/platform/ios/MGLMapView.mm
@@ -106,6 +106,7 @@ CLLocationDegrees MGLDegreesFromRadians(CGFloat radians)
     mbgl::Map *_mbglMap;
     MBGLView *_mbglView;
     mbgl::DefaultFileSource *_mbglFileSource;
+    BOOL _isWaitingForRedundantReachableNotification;
     
     NS_MUTABLE_ARRAY_OF(NSURL *) *_bundledStyleURLs;
 
@@ -244,6 +245,10 @@ std::chrono::steady_clock::duration secondsAsDuration(float duration)
                                                object:nil];
 
     MGLReachability* reachability = [MGLReachability reachabilityForInternetConnection];
+    if ([reachability isReachable])
+    {
+        _isWaitingForRedundantReachableNotification = YES;
+    }
     [reachability startNotifier];
 
     // setup annotations
@@ -415,9 +420,11 @@ std::chrono::steady_clock::duration secondsAsDuration(float duration)
 - (void)reachabilityChanged:(NSNotification *)notification
 {
     MGLReachability *reachability = [notification object];
-    if ([reachability isReachable]) {
+    if ( ! _isWaitingForRedundantReachableNotification && [reachability isReachable])
+    {
         mbgl::NetworkStatus::Reachable();
     }
+    _isWaitingForRedundantReachableNotification = NO;
 }
 
 - (void)dealloc

